### PR TITLE
Move block while display

### PR DESCRIPTION
**Pull request by itutu-tienday**

### CODE REVIEW

**Review by emuemuJP — APPROVED** [review]

File: frontend/src/components/FlowChart/ReactFlowComponent.tsx [hunk 1]
```diff
@@ -36,11 +36,29 @@ import {
...
+import { DialogContext } from 'components/Visualize/DialogContext'
+import { FileSelectDialog } from 'components/common/FileSelectDialog'
+import { FormHelperText, Popover } from '@mui/material'
+
+const initDialogFile = {
+  filePath: '',
+  open: false,
+  fileTreeType: '',
+  multiSelect: false,
+  onSelectFile: (_: any) => null,
```
Comment (ShogoAkiyama) [inline frontend/src/components/FlowChart/ReactFlowComponent.tsx:49]: Please, define type of variable,  not `any` type.

File: frontend/src/components/Visualize/DialogContext.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+export const DialogContext = createContext<{
+  onOpen: (nodeId: string) => any
+  onOpenDialogFile: (value: {
+    filePath: string | string[]
+    open: boolean
+    fileTreeType?: string
+    multiSelect: boolean
+    onSelectFile: (v: any) => any
+  }) => any
+  onMessageError: (v: { anchorElRef: { current: Element | null }; message: string }) => any
```
Comment (ShogoAkiyama) [inline frontend/src/components/Visualize/DialogContext.tsx:12]: Please, define type of variable,  not `any` type.

File: frontend/src/components/FlowChart/ReactFlowComponent.tsx [hunk 3]
```diff
@@ -117,27 +135,83 @@ export const ReactFlowComponent = React.memo<UseRunPipelineReturnType>(
...
-              nodeTypes={reactFlowNodeTypes}
-              edgeTypes={reactFlowEdgeTypes}
-              defaultPosition={[flowPosition.x, flowPosition.y]}
-              defaultZoom={flowPosition.zoom}
-              onMoveEnd={onMoveEnd}
+        <DialogContext.Provider
+          value={{
+            onOpen: setDialogNodeId,
+            onOpenDialogFile: setDialogFile as any,
+            onMessageError: setMessageError as any
```
Comment (ShogoAkiyama) [inline frontend/src/components/FlowChart/ReactFlowComponent.tsx:142]: Please, define type of variable, not any type.

File: frontend/src/components/FlowChart/ReactFlowComponent.tsx [hunk 4]
```diff
@@ -117,27 +135,83 @@ export const ReactFlowComponent = React.memo<UseRunPipelineReturnType>(
...
+              initialFilePath={dialogFile.filePath}
+              open={dialogFile.open}
+              onClickOk={(path) => {
+                dialogFile.onSelectFile(path)
+                setDialogFile(initDialogFile)
+              }}
+              onClickCancel={() => {
+                setDialogFile(initDialogFile)
+              }}
+              fileType={dialogFile.fileTreeType as any}
```
Comment (ShogoAkiyama) [inline frontend/src/components/FlowChart/ReactFlowComponent.tsx:185]: Please, define type of variable, not any type.

File: frontend/src/components/FlowChart/FlowChartNode/FileSelect.tsx [hunk 5]
```diff
@@ -97,7 +98,18 @@ export const FileSelectImple = React.memo<FileSelectImpleProps>(
...
         <ButtonGroup size="small" style={{ marginRight: 4 }}>
-          <Button variant="outlined" onClick={() => setOpen(true)}>
+          <Button
+            variant="outlined"
+            onClick={() => {
+              onOpenDialogFile({
+                open: true,
+                multiSelect,
+                filePath,
+                fileTreeType: fileTreeType as any,
```
Comment (ShogoAkiyama) [inline frontend/src/components/FlowChart/FlowChartNode/FileSelect.tsx:108]: Please, define type of variable, not any type.

File: frontend/src/components/FlowChart/FlowChartNode/AlgorithmNode.tsx [hunk 6]
```diff
@@ -39,7 +39,7 @@ import {
   NODE_RESULT_STATUS,
   RUN_STATUS,
 } from 'store/slice/Pipeline/PipelineType'
-import { AlgorithmOutputDialog } from './AlgorithmOutputDialog'
+import { DialogContext } from 'components/Visualize/DialogContext'
```
Comment (ShogoAkiyama) [inline frontend/src/components/FlowChart/FlowChartNode/AlgorithmNode.tsx:42]: Please, delete import library, which does not use.
```
src/components/FlowChart/FlowChartNode/AlgorithmNode.tsx
  Line 9:3:     'FormHelperText' is defined but never used    @typescript-eslint/no-unused-vars
  Line 17:8:    'Popover' is defined but never used           @typescript-eslint/no-unused-vars
  Line 326:10:  'open' is assigned a value but never used     @typescript-eslint/no-unused-vars
  Line 326:16:  'setOpen' is assigned a value but never used  @typescript-eslint/no-unused-vars
```

File: frontend/src/components/Visualize/DialogContext.tsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+}>({
+  onOpen: () => null,
+  onOpenDialogFile: () => null,
+  onMessageError: () => null,
+})
```
Comment (ShogoAkiyama) [inline frontend/src/components/Visualize/DialogContext.tsx:17]: please, change file path, not in `visualize` components but in `flowchart` components.
or create in store directory as redux.

**Comment by Reactplus-Hao** [comment]
@itutu-tienday 
上記のご指摘内容を対応して、下記のPRを作成致しました。
お手数ですが、ご確認お願い致します。
https://github.com/oist/optinist/pull/469

**Pull request merged**